pkg/scheduler/core/generic_scheduler.go

```diff
@@ -948,7 +948,9 @@ func (g *genericScheduler) selectVictimsOnNode(
 	var potentialVictims []*v1.Pod
 
 	removePod := func(rp *v1.Pod) error {
+		nodeInfoClone := nodeInfo.Clone()
 		if err := nodeInfo.RemovePod(rp); err != nil {
+			nodeInfo = nodeInfoClone
 			return err
 		}
 		status := prof.RunPreFilterExtensionRemovePod(ctx, state, pod, rp, nodeInfo)
```

Comment:
`unPreFilterExtensionRemovePod` maybe get different results for different pods
So if it returns error when removing a pod, but it may return `success` when remove others